src/main/java/io/divolte/server/ValidatedConfiguration.java

```diff
@@ -0,0 +1,535 @@
+/*
+ * Copyright 2015 GoDataDriven B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.divolte.server;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import javax.annotation.ParametersAreNonnullByDefault;
+import javax.annotation.ParametersAreNullableByDefault;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableList;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigException;
+import com.typesafe.config.ConfigList;
+import com.typesafe.config.ConfigValue;
+
+/**
+ * Container for a validated configuration loaded from a {@code Config}
+ * instance. This container allows access to the underlying configuration values
+ * through a {@link DivolteConfiguration} instance which can be obtained from
+ * calling {@link #configuration()}, only if the configuration is valid. This
+ * method throws an exception otherwise. checking the validity of the
+ * configuration must first be done through the {@link #isValid()} method. When
+ * the configuration is not valid, a list of {@code ConfigException} instances
+ * that were thrown during configuration parsing / loading is available by
+ * calling {@link #errors()}.
+ */
+@ParametersAreNonnullByDefault
+public final class ValidatedConfiguration {
+	private final static Logger logger = LoggerFactory.getLogger(ValidatedConfiguration.class);
+	
+    private final List<ConfigException> exceptions;
+    private final DivolteConfiguration divolteConfiguration;
+
+    /**
+	 * Creates an instance of a validated configuration. The underlying
```

Comment:
Oops. Had to re-install my Eclipse and throw away my workspaces and everything to get it back into shape again. Will re-configure it to use spaces.